internal/provider/kubernetes/kubernetes.go

```diff
@@ -37,18 +53,19 @@ func New(cfg *rest.Config, svr *config.Server) (*Provider, error) {
 	}
 
 	// Create and register the controllers with the manager.
-	if err := newGatewayClassController(mgr, svr); err != nil {
+	if err := newGatewayClassController(mgr, svr, resourceTable); err != nil {
 		return nil, fmt.Errorf("failed to create gatewayclass controller: %w", err)
 	}
-	if err := newGatewayController(mgr, svr); err != nil {
+	if err := newGatewayController(mgr, svr, resourceTable); err != nil {
 		return nil, fmt.Errorf("failed to create gateway controller: %w", err)
 	}
 	// TODO: Add httproute controllers.
 	// xref: https://github.com/envoyproxy/gateway/issues/163
 
 	return &Provider{
-		manager: mgr,
-		client:  mgr.GetClient(),
+		manager:       mgr,
+		client:        mgr.GetClient(),
+		resourceTable: resourceTable,
```

Comment:
nit: is `resourceTable` it needed to be saved in the Provider ctx ?, since it is being passed to `New`, the caller should already have access to it

Reply:
nvm I see `type ResourceTable = kubernetes.ResourceTable` which works